components/DateInput.vue

```diff
@@ -0,0 +1,169 @@
+        <input
+          :class="inputClass"
+          aria-describedby="dayinput"
+          :id="`${uniqueId}-${criteriaKey}-day`"
+          :name="`${uniqueId}-${criteriaKey}-day`"
+          type="text"
+          maxlength="2"
+          pattern="[0-9]*"
+          inputmode="numeric"
+          :value="pullDateValue(dateResponse, 1)"
+          @change="updateElibilityDate($event, criteriaKey)"
+        />
+      </div>
+      <div class="usa-form-group usa-form-group--year">
+        <label 
+          :class="labelClass" 
+          :for="`${uniqueId}-${criteriaKey}-year`">Year</label>
+        <input
+          :class="inputClass"
+          aria-describedby="yearinput"
+          :id="`${uniqueId}-${criteriaKey}-year`"
+          :name="`${uniqueId}-${criteriaKey}-year`"
+          type="text"
+          minlength="4"
+          maxlength="4"
+          pattern="[0-9]*"
+          inputmode="numeric"
+          :value="pullDateValue(dateResponse, 2)"
+          @change="updateElibilityDate($event, criteriaKey)"
+        />
+      </div>
+    </div>
+  </fieldset>
+</template>
+
+<style scoped>
+.usa-legend {
+  margin-bottom: -0.6875rem;
+}
+
+.usa-legend--empty,
+.usa-label--empty,
+.usa-input--empty {
+  color: #1b1b1b;
+}
+
+.usa-legend--success,
+.usa-label--success,
+.usa-input--success {
+  color: green;
+  font-weight: bold;
+}
+
+.usa-legend--error,
+.usa-label--error,
+.usa-input--error {
+  color: red;
+  font-weight: bold;
+}
+</style>
+
+<script>
+import _ from 'lodash'
+export default {
+    name: 'DateInput',
+    props: {
+        criteriaKey: {
+            type: String,
+            default: 'no criteria key provided'
+        },
+        label: {
+            type: String,
+            default: 'no label provided'
+        },
+        response: {
+            type: [String, Object, Boolean],
+            default: 'no response provided'
+        },
+        dateResponse: {
+          type: String,
+          default: 'no response inputted'
+        },
+        location: {
+          type: String,
+          default: "benefit-card",
+          validator: (value) => { return ['benefit-card', 'left-rail'].includes(value) }
+        }
+    },
+    data() {
+        return {
+            uniqueId: _.uniqueId('dateinput-')
+        }
+    },
+    computed: {
+      selectedStyle() {
+        if (this.location === "left-rail") {
+          return "text-bold"
+        }
+        return null
+      },
+      labelClass() {
+        return `usa-label usa-label--${this.response ? 'success': 
+            this.response == null ? 'empty': 'error'}`
+      },
+      legendClass() {
+        return `usa-legend usa-legend--${this.response ? 'success': this.response == null ? 'empty': 'error'} ${this.selectedStyle}`
+      },
+      inputClass() {
+        return `usa-input usa-input--${this.response ? 'success': 
+            this.response == null ? 'empty': 'error'}`
+      }
+    },
```

Comment:
I originally had them here because of the Vue lifecycle they would be called when properties are updated and that would cause a change in the style of the input